javascript/javascript3/week3/script.js

```diff
@@ -0,0 +1,240 @@
+let counter = 0;
+function intervalCircles(){
+    const interv = setInterval(()=>{
+        counter++;
+        const c = new Circle(randomInt(0,500),randomInt(0,500),randomInt(5,30),0,360,getRandomColor(),ctx2);
+        c.draw();
+        if(counter >= maxCircles){
+            clearInterval(interv);
+        }
+    },100)
+}
+
+intervalCircles();
+
+//Classmates
+const classmates = document.getElementById("classmates");
+
+function renderRepositories(username){
+  const getData = new Promise((resolve,reject) =>{
+    fetch(`https://api.github.com/search/repositories?q=user:${username}`).then((msg)=>{
+      return msg.json();
+    }).then((msg)=>{
+      resolve(msg);
+    }).catch((e)=>{
+      reject(e);
+    });
+  });
+  
+  //Fullname, url and owner
+  const userDiv = document.createElement("div");
+  const repOwner = document.createElement("h2");
+  const br = document.createElement("br");
+  repOwner.innerHTML = "Github User: " + username;
+  userDiv.appendChild(repOwner);
+  classmates.appendChild(userDiv);
+  userDiv.classList.add("user-div");
+  
+  getData.then((data)=>{
+    data.items.forEach((rep)=>{
+      const repDiv = document.createElement("div");
+      userDiv.appendChild(repDiv);
+
+      const repName = document.createElement("p");
+      repName.innerHTML = rep.full_name;
+      repDiv.appendChild(repName);
+
+      const repUrl = document.createElement("a");
+      repUrl.href = rep.url;
+      repUrl.innerHTML = "Link";
+      repDiv.appendChild(repUrl);
+
+    })
+  });
+}
+
+renderRepositories("");
+
+//Shopping cart
+const productsDiv = document.getElementById("products");
+const productsPrice = document.getElementById("price");
+const productsUsername = document.getElementById("username");
+
+const APIKEY = "9f2a80d8316b32f573e9";
+
+class Product {
+  constructor(name, price) {
+    this.name = name;
+    this.price = price;
+  }
+
+  convertToCurrency(currency){
+    return new Promise((resolve,reject)=>{
+        fetch(`https://free.currconv.com/api/v7/convert?q=DKK_${currency}&compact=ultra&apiKey=${APIKEY}`).then((msg)=>{
+          return msg.json();  
+        }).then((msg)=>{
+            resolve(msg[Object.keys(msg)[0]] * this.price);
+        }).catch((e)=>{
+            reject(e);
+        });
+    });
+  }
+
+}
+
+class ShoppingCart {
+  constructor() {
+    this.products = [];
+  }
+
+  addProduct(product) {
+    this.products.push(product);
+  }
+
+  removeProduct(product) {
+    let counter = 0;
+    let foundProduct = false;
+
+    this.products.forEach((p)=>{
+        if(p==product){
+            foundProduct = true;
+            return;
+        }
+        counter++;
+    });
+    if(foundProduct){
+        this.products.splice(counter,1);
+    }
+  }
+
+  searchProduct(productName) {
+    let foundProducts = [];
+    this.products.forEach((p)=>{
+        if(p.name == productName){
+            foundProducts.push(p);
+        }
+    });
+    return foundProducts;
+  }
+
+  getTotal() {
```

Comment:
You could simplify this by using [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce).